queries/cdmq/README.md

```diff
@@ -0,0 +1,451 @@
+# cdmq
+
+## Introduction
+The contents of this directory contain a collection of scripts in Javascript intended to be executed with [node.js](https://nodejs.org).  These scripts get data from an elasticsearch instance.  The data must be in Common Data Format. which is documented in this project under [templates](../templates).  The scripts here are meant to help inspect, compare, and export data from benchmarks and performance & resource-utilization tools, in order to report and investigate performance.
+
+In order to generate this data, you must run a benchmark via automation framework which uses the Common Data Format and index that data into elasticsearch.  One of those automation frameworks is the [crucible](https://github.com/perftool-incubator/crucible) project.  A subproject of crucible, [crucible-examples](https://github.com/perftool-incubator/crucible-examples), includes scenarios to run some of these benchmarks.
+
+## Terms
+Many of the scripts refer to different terms we associate with either running a benchmark or examining the resulting data, and these terms are not always universally known or agreed upon for specific benchmarks (like uperf and fio), or even benchmark automation frameworks,  Nevertheless, the CommonDataModel project has adopted the following terms, which originate from the crucible project.  You will need to become familiar with these terms in order to use these scripts:
+
+- benchmark: A specific benchmark, like fio, uperf, trafficgen, or oslat.
+- run: An invocation of a command, like `crucible run`, which facilitates the execution of a benchmark, often running the benchmark many times.  In the context `cdmq`, this usually refers to the data that was generated for that run.
+- iteration: A set of parameters to execute a benchmark, for example, for uperf:  `test-type: stream, wsize: 256, nthreads: 16, duration: 90`.  One or more unique iterations typically make up a run
+- parameter: An option used for the underlying benchmark.  Most parameters are unique to a specific benchmark.
+- sample: An actual execution of an iteration.  Often there will be multiple samples for an iteration, in order to provide an average and standard-deviation.
+- period: A time-period for a sample.  When a sample is executed, there may be one or more periods which represent a certain phase for the benchmark, like warmup, measurement, etc.
+- primary-period: A period where a benchmark's primary metric is measured
+- primary-metric: A benchmark's most common metric, like `Gbps`, or `IOPS`.  Each iteration has a primary-metric, but different iterations (different combinations of parameters) might have a different primary-metric.  For example uperf samples with `test-type: stream` have a primary metric of `Gbps`, while `test-type: rr` uses `transactions-sec`
+- metric: Some unit of measure, either a measure of throughput (work/time) or a "count" (elapsed-time, latency, level, occupancy, etc), or a simple "pass/fail"
```

Comment:
given what you are trying to do with the compare script, should "tag" be defined here?

Reply:
probably.  I'll add it with the compare PR